client/components/modal.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import AppContext from '../lib/app-context';
+
+export default class PostModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      postPicture: 'placeholder.png',
+      location: '',
+      caption: ''
+    };
+    this.fileInputRef = React.createRef();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleCreatePost = this.handleCreatePost.bind(this);
+  }
+
+  handleChange(event) {
+    if (event.target.getAttribute('type') === 'file') {
+      this.setState({
+        postPicture: URL.createObjectURL(event.target.files[0])
+      });
+    } else {
+      const { name, value } = event.target;
+      this.setState({ [name]: value });
+    }
+  }
+
+  handleCreatePost(event) {
+    event.preventDefault();
+    const userId = this.context.user.userId;
```

Comment:
Instead of trusting the client to tell you its `userId`, the server should expect that the request includes a token (with authorization middleware). The token is the source of truth for identifying the user.